### PR TITLE
Replace fence tests with new onSubmittedWorkDone tests

### PR DESCRIPTION
**Pull request by kainino0x**
Please review for any things about onSubmittedWorkDone that you think should be tested but are not.

<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented in `helper_index.md`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)
    
**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [x] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [x] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [x] Existing (or new) test helpers are used where they would reduce complexity.
- [x] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

File: src/webgpu/api/operation/onSubmittedWorkDone.spec.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Note that any promise timeouts will be detected by the framework.
+`;
+
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { range } from '../../../common/framework/util/util.js';
+import { GPUTest } from '../../gpu_test.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('without_work')
```
Comment (Kangz) [inline src/webgpu/api/operation/onSubmittedWorkDone.spec.ts:13]: We should also check the ordering of onSubmittedWorkDone with respect to each other, and with respect to mapAsync callbacks.

**Review by Kangz — APPROVED** [review]
LGTM

File: src/webgpu/api/operation/onSubmittedWorkDone.spec.ts (hunk 1 above)
Comment (kainino0x) [reply src/webgpu/api/operation/onSubmittedWorkDone.spec.ts:13]: Latter is tracked here https://github.com/gpuweb/cts/blob/main/src/webgpu/api/operation/async_ordering/README.txt

**Review by Kangz — APPROVED** [review]
Still LGTM

**Review by toji — APPROVED** [review]

**Pull request merged**